pkg/rancher-desktop/backend/lima.ts

```diff
@@ -1643,7 +1639,20 @@ export default class LimaBackend extends events.EventEmitter implements VMBacken
     }
   }
 
-  copyFileIn(hostPath: string, vmPath: string): Promise<void> {
+  async copyFileIn(hostPath: string, vmPath: string): Promise<void> {
+    // TODO This logic is copied from writeFile() above and should be simplified.
+    const workdir = await fs.promises.mkdtemp(path.join(os.tmpdir(), `rd-${ path.basename(hostPath) }`));
+    const tempPath = `/tmp/${ path.basename(workdir) }.${ path.basename(hostPath) }`;
+
+    try {
+      await this.lima('copy', hostPath, `${ MACHINE_NAME }:${ tempPath }`);
+      await this.execCommand('chmod', '644', tempPath);
+      await this.execCommand({ root: true }, 'mv', tempPath, vmPath);
+    } finally {
+      await fs.promises.rm(workdir, { recursive: true });
+      await this.execCommand({ root: true }, 'rm', '-f', tempPath);
+    }
+
     return this.lima('copy', hostPath, `${ MACHINE_NAME }:${ vmPath }`);
```

Comment:
Why are we running `copy` twice? If this succeeds, why did we need to be root?

Reply:
Good catch; that line should have been deleted. I'm curious why it didn't still throw an error. Anyways, it is gone now.